### PR TITLE
GenerateIdTransformer with SEQUENCE and ROWID modes.

### LINKED ISSUE #413: Make GenerateIdTransformer use row ID.
It would be nice to have an option to generate an ID that will preserve the order of the records in a file. Adding a mode to GenerateIdTransformer that will return the row ID should do the trick.

### PR DESCRIPTION
**Pull request by tomaszguzialek**
Fixes #413

### CODE REVIEW

File: components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java [hunk 1]
```diff
@@ -38,35 +38,48 @@
...
+
+    @Override
+    public OutputColumns getOutputColumns() {
+        return new OutputColumns(Integer.class, "Generated ID");
+    }
+
+    @Override
+    public Integer[] transform(InputRow inputRow) {
+        int id;
+        if (IdType.ROWID.equals(idType)) {
```
Comment (kaspersorensen) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java]: when doing enum comparison the double-equal operator (==) is a bit nicer to use :)

File: components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java [hunk 2]
```diff
@@ -38,35 +38,48 @@
 @Categorized(NumbersCategory.class)
 public class GenerateIdTransformer implements Transformer {
 
-	@Configured
-	@Description("A column which represent the scope for which the ID will be generated. "
-			+ "If eg. a source column is selected, an ID will be generated for each source record. "
-			+ "If a transformed column is selected, an ID will be generated for each record generated that has this column.")
-	InputColumn<?> columnInScope;
+    public enum IdType {
+        SEQUENCE, ROWID;
```
Comment (kaspersorensen) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java]: ROW_NUMBER would be a bit better than ROWID in my opinion

File: components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java [hunk 3]
```diff
@@ -38,35 +38,48 @@
...
+    }
+
+    @Override
+    public OutputColumns getOutputColumns() {
+        return new OutputColumns(Integer.class, "Generated ID");
+    }
+
+    @Override
+    public Integer[] transform(InputRow inputRow) {
+        int id;
```
Comment (kaspersorensen) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java]: final? :smiling_imp:

File: components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java (hunk 2 above)
Comment (kaspersorensen) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/numbers/GenerateIdTransformer.java]: And if you make your enum also implement HasName then they can have pretty names in the UI and documentation :-)

**Comment by kaspersorensen** [comment]
Not sure if I was being too evil? :hurtrealbad:

**Comment by tomaszguzialek** [comment]
I am happy to see very detailed comments, so you can keep "evil-boss-final-keyword-checkstyle" mode on :). Comments addressed, I am merging it when Travis agrees.

**Pull request merged**